### PR TITLE
Add support for --token-file-path connection option

### PR DESCRIPTION
**Pull request by sfc-gh-turbaszek**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [ ] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on MacOS.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually on Windows.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Newest version of Python connector added option for passing `token_file_path` as connection option.
https://github.com/snowflakedb/snowflake-connector-python/blob/42fa6ebe9404e0e17afdacfcaceb311dda5cde3e/DESCRIPTION.md?plain=1#L12

### CODE REVIEW

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -22,6 +22,8 @@
 ## New additions
 * Added support for `title` field in Streamlit definition in `snowflake.yml` project file.
 * Added `--auto-compress` flag to `snow stage copy` command enabling use of gzip to compress files during upload.
+* Added connection option `--token-file-path` allowing passing OAuth token using a file. The function is also
+  supported by setting `token_file_path` in connection definition.
```
Comment (sfc-gh-astus) [inline RELEASE-NOTES.md:26]: Do we want to permission checks like config.toml has?

Comment (sfc-gh-turbaszek) [reply RELEASE-NOTES.md:26]: Why should we? What scenario we see as risky? If the input token is invalid then Snowflake will not authorize the connection.

Comment (sfc-gh-astus) [reply RELEASE-NOTES.md:26]: You are right.

**Review by sfc-gh-astus — APPROVED** [review]

**Review by sfc-gh-astus — APPROVED** [review]

File: RELEASE-NOTES.md [hunk 2]
```diff
@@ -44,6 +44,8 @@
     * vars from `--env` command line argument.
     * vars from shell environment variables.
     * vars from `env` section of project definition file.
+* Added connection option `--token-file-path` allowing passing OAuth token using a file. The function is also
+  supported by setting `token_file_path` in connection definition.
```
Comment (sfc-gh-astus) [inline RELEASE-NOTES.md:48]: Move it to 2.7.0

**Review by sfc-gh-mraba — APPROVED** [review]

**Pull request merged**